src/injected/web/requests.js

```diff
@@ -112,7 +119,59 @@ function start(req, id) {
 }
 
 function getFullUrl(url) {
-  const a = document.createElementNS(NS_HTML, 'a');
-  a.setAttribute('href', url);
-  return a.href;
+  const a = document::createElementNS(NS_HTML, 'a');
+  a::setAttribute('href', url);
+  return a::getHref();
+}
+
+const { keys, getAll } = FormData.prototype;
+const { FileReader } = global;
+const { readAsArrayBuffer } = FileReader.prototype;
+
+async function encodeBody(body) {
+  const cls = getType(body);
+  switch (cls) {
+  case 'formdata': {
+    const data = {};
+    const resolveKeyValues = async (key) => {
+      const values = body::getAll(key)::map(encodeBody);
```

Comment:
Also `map` here is incorrect.

Reply:
Why? it's what the original code did.

Reply:
The original code is:
```js
const values = array.map(xxx);
```
Now:
```js
const values = array::map(xxx);
```
Which is the same as:
```js
const values = map.call(array, xxx);
```
`map` is different here! Quite confusing I think. I didn't figure it out until failed running my test script.

Reply:
Renamed the variable to `idMap`.